src/arch/riscv/inc/arch/platform.h

```diff
@@ -8,8 +8,23 @@
 
 #include <bao.h>
 
+#define PLIC  (1)
+#define APLIC (2)
+
 struct arch_platform {
-    paddr_t plic_base;
+    union {
+        struct {
+            paddr_t base;
+        } plic;
+        struct {
+            struct {
+                paddr_t base;
+            } aplic;
+            struct {
+                paddr_t base;
+            } imsic;
```

Comment:
Just noticed that I have already introduced the imsic. Should I remove it from this PR and introduce it in IMSIC PR?

Reply:
Yes. Please do.

Reply:
Done!